### PR TITLE
feat: add tools table for creating and deleting tools

### PR DESCRIPTION
**Pull request by tylerslaton**
https://github.com/user-attachments/assets/a7d59f4c-2c98-4df0-8a22-ec1da8e1ef87

### CODE REVIEW

File: ui/admin/app/components/tools/CreateTool.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            });
+            reset();
+            onSuccess();
+        } catch (error) {
+            console.error("Failed to create tool reference:", error);
+        } finally {
+            setIsSubmitting(false);
+        }
+    };
+
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/tools/CreateTool.tsx]:
```ts
    const { execute: onSubmit, isLoading } = useAsync(
        async (data: CreateToolReference) => {
            await ToolReferenceService.createToolReference({
                toolReference: { ...data, toolType: "tool" },
            });
            reset();
            onSuccess();
        },
        {
            onError: (error) =>
                console.error("Failed to create tool reference:", error),
        }
    );
```

File: ui/admin/app/components/tools/ToolsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                            onDelete();
+                            setIsDeleteDialogOpen(false);
+                        }}
+                    >
+                        Delete
+                    </Button>
+                </DialogFooter>
+            </DialogContent>
+        </Dialog>
+    );
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/tools/ToolsTable.tsx]: I created a helper component called `ConfirmationDialog` that will handle this quite nicely

File: ui/admin/app/routes/_auth.tools._index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    DialogTitle,
+    DialogTrigger,
+} from "~/components/ui/dialog";
+import { Input } from "~/components/ui/input";
+
+export default function Tools() {
+    const { data: tools, mutate } = useSWR(
+        ToolReferenceService.getToolReferences.key("tool"),
+        () => ToolReferenceService.getToolReferences("tool")
+    );
```
Comment (ryanhopperlowe) [inline ui/admin/app/routes/_auth.tools._index.tsx:33]: nit: preload in a `clientLoader`?

File: ui/admin/app/components/tools/CreateTool.tsx (hunk 1 above)
Comment (tylerslaton) [reply ui/admin/app/components/tools/CreateTool.tsx]: 🤦🏻 So much better, thanks Ryan.

File: ui/admin/app/components/tools/ToolsTable.tsx (hunk 2 above)
Comment (tylerslaton) [reply ui/admin/app/components/tools/ToolsTable.tsx]: Nice, yep does this exactly.

File: ui/admin/app/routes/_auth.tools._index.tsx (hunk 3 above)
Comment (tylerslaton) [reply ui/admin/app/routes/_auth.tools._index.tsx:33]: Done!

**Pull request merged**